#data_dir: "/media/midea/eai-53/1013/02"
#data_dir: "/media/midea/Extreme SSD/folding_skirt_0folded_sj_20251229"
# data_dir: "/media/midea/Extreme SSD/random_folding_pants_basket_1folded_xdyna_sj_20251231_1"
#data_dir: "/media/midea/Extreme SSD/random_folding_skirt_basket_2folded_sj_20251230"
#data_dir: "/media/midea/Extreme SSD/folding_trousers_0folded_sj_20251226"
#data_dir: "/home/midea/random_folding_shirts_basket_1folded_xdyna_sj_20251217_prac"
#data_dir: "/media/midea/Extreme SSD/random_folding_shirts_wooden_basket_1folded_xdyna2_dhl_20251022"
data_dir: "/home/r1lite/GalaxeaDataset/20260104"
max_queue_size: 5
max_age: 0.5
collection_frequency: 50

datasets:
  "/subtask":
    topic: "/pedal_status"
    processor: "data_collector.data_processor.PedalProcessor"
    processor_config:
      output_shape: [1]
      dtype: "int32"
      description: "Pedal status converted to subtask ID: step=10, double step=100, normal=0"

  "/observations/images/cam_right_wrist":
    # topic: "/cam_right_wrist/color/image_rect_raw"
    topic: "/hdas/camera_wrist_right/color/image_rect_raw"
    processor: "data_collector.data_processor.ImageProcessor"
    processor_config:
      resize: [640, 480]
      output_shape: [480, 640, 3]
      dtype: "uint8"
      description: "Right wrist camera image"

  "/observations/images/cam_left_wrist":
    # topic: "/cam_left_wrist/color/image_rect_raw"
    topic: "/hdas/camera_wrist_left/color/image_rect_raw"
    processor: "data_collector.data_processor.ImageProcessor"
    processor_config:
      resize: [640, 480]
      output_shape: [480, 640, 3]
      dtype: "uint8"
      description: "Left wrist camera image"

  # "/observations/images/cam_high":
  #   topic: "/cam_d457/color/image_raw"
  #   processor: "data_collector.data_processor.ImageProcessor"
  #   processor_config:
  #     resize: [640, 480]
  #     output_shape: [480, 640, 3]
  #     dtype: "uint8"
  #     description: "Top-mounted D457 camera image"


topics:
  "/pedal_status":
    msg_type: "sensor_msgs.msg.Joy"
    description: "Pedal input device"

  "/cam_right_wrist/color/image_rect_raw":
    msg_type: "sensor_msgs.msg.Image"
    description: "Right wrist camera"

  "/cam_left_wrist/color/image_rect_raw":
    msg_type: "sensor_msgs.msg.Image"
    description: "Left wrist camera"

    # "/cam_high/color/image_rect_raw":
    # msg_type: "sensor_msgs.msg.Image"
    # description: "high camera"


  # "/cam_front/color/image_rect_raw":
  #   msg_type: "sensor_msgs.msg.Image"
  #   description: "Front camera between arms"


  # "/cam_l515/color/image_raw":
  #   msg_type: "sensor_msgs.msg.Image"
  #   description: "Top-mounted L515 camera"

  # "/cam_l515/depth/image_rect_raw":
  #   msg_type: "sensor_msgs.msg.Image"
  #   description: "Top-mounted L515 depth image"

  # "/cam_d457/color/image_raw":
  #   msg_type: "sensor_msgs.msg.Image"
  #   description: "Top-mounted D457 camera"
